dart/flat_buffers.dart

```diff
@@ -0,0 +1,1095 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:collection';
+import 'dart:convert';
+import 'dart:math';
+import 'dart:typed_data';
+
+/**
+ * Reader of lists of boolean values.
+ *
+ * The returned unmodifiable lists lazily read values on access.
+ */
+class BoolListReader extends Reader<List<bool>> {
```

Comment:
why are the ones for bools singled out here?

Reply:
This is a special case where multiple bools are stored as a single uint8.  It's nice for space savings if you have a lot of bools, but it's not interoperable with other langauges and so should probably be removed.

Reply:
Yes, we tried to pack information as much as possible, because we wanted at some point to put it into Dart SDK, but now we don't. Mostly it is for indexes, see https://github.com/dart-lang/sdk/blob/master/pkg/analyzer/lib/src/summary/idl.dart#L280

It will not be the end of the world (at least not worse than migrating to a slightly different flat buffers implementation) is this goes away.

Reply:
Generating binaries that are 100% compliant with the other implementations is by far the most important concern of all. So yes, please remove this special case. In fact, as part of the test, also reading the binary generated by C++ is a small step in helping to verify that.

Reply:
Fixed this. I like the way they implemented that, unfortunately it's non-standard.